src/components/pages/rules/flyouts/ruleEditor/ruleEditor.js

```diff
@@ -159,10 +160,21 @@ export class RuleEditor extends LinkedComponent {
             },
             error => this.setState({ error, isPending: false, changesApplied: true })
           );
+        logEvent(this.ruleToDiagnosticsModel(requestProps, 'apply_click'));
       }
     }
   }
 
+  ruleToDiagnosticsModel = (rule, eventName) => {
+    const eventProps = {
+      device_group: rule.groupId,
```

Comment:
The ruleEditor (a UI component) should have no knowledge of the actual backend model structures. Move this to a services/models file instead.

Reply:
fixed